src/functions.rs

```diff
@@ -450,6 +519,43 @@ pub mod stdlib {
     pub mod string {
         use super::*;
 
+        /// The implementation of the standard [`format`][`crate::reference::functions#format`] function.
+        pub struct Format;
+
+        impl Function for Format {
+            fn call(
+                &self,
+                _graph: &mut Graph,
+                _source: &str,
+                parameters: &mut dyn Parameters,
+            ) -> Result<Value, ExecutionError> {
+                let format = parameters.param()?.into_string()?;
+                let mut result = String::new();
+                let mut it = format.chars().enumerate().into_iter();
+                while let Some((_, c)) = it.next() {
+                    match c {
+                        '{' => match it.next() {
+                            Some((_, '{')) => result.push('{'),
+                            Some((_, '}')) => {
+                                let value = parameters.param()?;
+                                result += &value.to_string();
+                            },
+                            Some((i, c)) => return Err(ExecutionError::FunctionFailed("format".into(), format!("Unexpected character `{}` after `{{` at position {} in format string `{}`. Expected `{{` or `}}`.", c, i + 1, format))),
+                            None => return Err(ExecutionError::FunctionFailed("format".into(), format!("Unexpected end of format string `{}` after `{{`. Expected `{{` or `}}`.", format))),
+                        },
+                        '}' => match it.next() {
+                            Some((_, '}')) => result.push('}'),
+                            Some((i, c)) => return Err(ExecutionError::FunctionFailed("format".into(), format!("Unexpected character `{}` after `}}` at position {} in format string `{}`. Expected `}}`.", c, i + 1, format))),
+                            None => return Err(ExecutionError::FunctionFailed("format".into(), format!("Unexpected end of format string `{}` after `{{. Expected `}}`.", format))),
+                        },
+                        c => result.push(c),
+                    }
+                }
```

Comment:
If we ever want to support modifiers in the format strings, there's a [`runtime_fmt`](https://docs.rs/runtime-fmt/latest/runtime_fmt/) crate that might be useful.  No need to bring that in yet, though, this is :+1: for the simplified format strings that you're currently supporting.

Reply:
Thanks for that reference! I was looking for something like that and couldn't find it. It was indeed the purpose to be open to richer specifiers in the future.